dpki/src/keystore.rs

```diff
@@ -0,0 +1,330 @@
+use crate::{
+    key_bundle::KeyBundle,
+    keypair::generate_random_sign_keypair,
+    seed::{generate_random_seed_buf, IndexedSeed, RootSeed, SeedContext, SeedTrait, SeedType},
+    utils, SEED_SIZE,
+};
+use holochain_core_types::{
+    agent::Base32,
+    cas::content::Address,
+    error::{HcResult, HolochainError},
+    signature::Signature,
+};
+
+use holochain_sodium::{kdf, secbuf::SecBuf};
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+pub enum Secret {
+    Key(KeyBundle),
+    RootSeed(RootSeed),
+    IndexedSeed(IndexedSeed),
+}
+
+#[allow(dead_code)]
+struct Keystore {
+    keys: HashMap<String, Arc<Mutex<Secret>>>,
+}
+
+impl Keystore {
+    #[allow(dead_code)]
+    pub fn new() -> Self {
+        Keystore {
+            keys: HashMap::new(),
+        }
+    }
+
+    /// return a list of the identifiers stored in the keystore
+    #[allow(dead_code)]
+    pub fn list(&self) -> Vec<String> {
+        self.keys.keys().map(|k| k.to_string()).collect()
+    }
+
+    /// adds a random root seed into the keystore
+    #[allow(dead_code)]
+    pub fn add_random_seed(&mut self, id_str: &str, size: usize) -> HcResult<()> {
+        let id = id_str.to_string();
+        if self.keys.contains_key(&id) {
+            return Err(HolochainError::ErrorGeneric(
+                "identifier already exists".to_string(),
+            ));
+        }
+        let seed_buf = generate_random_seed_buf(size);
```

Comment:
the other functions use `SEED_SIZE` should we simplify this api?

Reply:
well actually, the reason I did this is because in https://hackmd.io/tVy22HhJQPuHO28gtD_UTQ?both you have the "size" param for `keystore_new_random` so I made the generalized function in keystore to use that.  If we don't need to give the end-user access to the size of the random seed, then I'll take it out.